### PR TITLE
Check if important files are listed in `.gitignore`

### LINKED ISSUE #41: How to deal with `cran-comments.md` listed in `.gitignore`
It just happened to me that a repo had `cran-comments.md`in `.gitignore`. Should we attempt to force add here?

Otherwise `pre_release()` fails.

### PR DESCRIPTION
**Pull request by pat-s**
fixes #41 

Which files should we check for besides?

### CODE REVIEW

**Comment by pat-s** [comment]
ping @krlmlr

File: R/auto.R [hunk 1]
```diff
@@ -291,6 +293,17 @@ check_post_release <- function() {
...
-  if (out == "") return(character())
+  if (out == "") {
+    return(character())
+  }
   strsplit(out, ", *")[[1]]
 }
+
+check_gitignore <- function(files) {
+  files_ignored <- map_chr(files, ~ system2("git", c("check-ignore", .x), stdout = TRUE))
+  if (length(files_ignored >= 1)) {
```
Comment (krlmlr) [inline R/auto.R:304]: What are we checking here?

File: R/auto.R [hunk 2]
```diff
@@ -291,6 +293,17 @@ check_post_release <- function() {
...
   out <- attr(gh::gh("/user"), "response")$"x-oauth-scopes"
-  if (out == "") return(character())
+  if (out == "") {
+    return(character())
+  }
   strsplit(out, ", *")[[1]]
 }
+
+check_gitignore <- function(files) {
+  files_ignored <- map_chr(files, ~ system2("git", c("check-ignore", .x), stdout = TRUE))
```
Comment (krlmlr) [inline R/auto.R:303]: Running `check-ignore -q` without `stdout = TRUE` seems slightly safer.

Comment (pat-s) [reply R/auto.R:303]: The return value of `check-ignore` is a bit ugly. 

Both

`system2("git", c("check-ignore", "cran-comments.md"), stdout = FALSE)` and 
`system2("git", c("check-ignore","-q", "cran-comments.md"), stdout = FALSE)` 

return nothing if `cran-comments.md` is in `.gitignore`.

However, what we want is to detect this case.

Hence, we call `system2("git", c("check-ignore", "cran-comments.md"), stdout = TRUE)` which returns `"cran-comments.md"`. 

I do not see how `system2("git", c("check-ignore","-q", "cran-comments.md"), stdout = FALSE)` should work here?

File: R/auto.R (hunk 1 above)
Comment (pat-s) [reply R/auto.R:304]: `files_ignored` stores names of files which are listed in `.gitignore` but shouldn`t be there.

File: R/auto.R (hunk 2 above)
Comment (krlmlr) [reply R/auto.R:303]: Without `stdout = FALSE`, an exit code is returned (invisibly).

```suggestion
  files_ignored <- map_int(files, ~ system2("git", c("check-ignore", "-q", .x), stdout = FALSE))
```

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**